test/ui.spec.ts

```diff
@@ -94,279 +146,1444 @@ describe('Signer integration', function () {
+        BroadcastedTx<SignedTx<SignerInvokeTx>>
+      ];
+
+      const [parsedApproveResult] = result;
+      const expectedApproveResult = {
+        type: 16 as const,
+        version: 2,
+        senderPublicKey: issuer.publicKey,
+        dApp: data.dApp,
+        call: data.call,
+        payment: data.payment as NonNullable<typeof data.payment>,
+        fee: 500000,
+        chainId,
+      };
+
+      const bytes = makeTxBytes({
+        ...expectedApproveResult,
+        timestamp: parsedApproveResult.timestamp,
+      });
+
+      expect(parsedApproveResult).to.deep.contain(expectedApproveResult);
+      expect(parsedApproveResult.id).to.equal(base58Encode(blake2b(bytes)));
+
+      expect(
+        verifySignature(issuer.publicKey, bytes, parsedApproveResult.proofs[0])
+      ).to.be.true;
     });
 
-    it('with no args but many payments', async function () {
-      await signedTxShouldBeValid.call(
-        this,
-        'invoke',
-        INVOKE_NO_ARGS_MANY_PAYMENTS,
-        By.xpath("//div[contains(@class, 'scriptInvocation-transaction')]")
-      );
+    it('Invoke with argument', async function () {
+      const data: InvokeArgs = {
+        dApp: user1.address,
+        call: {
+          function: 'withdraw',
+          args: [{ type: 'integer', value: 100 }],
+        },
+        payment: [],
+      };
+
+      await performInvokeTransaction.call(this, data);
+
+      await approveMessage.call(this);
+      await closeMessage.call(this);
+
+      const result = (await getApproveResult.call(this)) as [
+        BroadcastedTx<SignedTx<SignerInvokeTx>>
+      ];
+
+      const [parsedApproveResult] = result;
+      const expectedApproveResult = {
+        type: 16 as const,
+        version: 2,
+        senderPublicKey: issuer.publicKey,
+        dApp: data.dApp,
+        call: data.call,
+        payment: data.payment as NonNullable<typeof data.payment>,
+        fee: 500000,
+        chainId,
+      };
+
+      const bytes = makeTxBytes({
+        ...expectedApproveResult,
+        timestamp: parsedApproveResult.timestamp,
+      });
+
+      expect(parsedApproveResult).to.deep.contain(expectedApproveResult);
+      expect(parsedApproveResult.id).to.equal(base58Encode(blake2b(bytes)));
+
+      expect(
+        verifySignature(issuer.publicKey, bytes, parsedApproveResult.proofs[0])
+      ).to.be.true;
     });
 
-    it('with native args and no payments', async function () {
-      await signedTxShouldBeValid.call(
-        this,
-        'invoke',
-        INVOKE_NATIVE_ARGS_NO_PAYMENTS,
-        By.xpath("//div[contains(@class, 'scriptInvocation-transaction')]")
-      );
+    it('Invoke with long arguments and payments list', async function () {
+      const binLong =
+        'base64:' +
+        btoa(
+          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+          // @ts-ignore
+          new Uint8Array(Array(100).fill([0, 1, 2, 3, 4, 5, 6, 7, 8, 9]).flat())
```

Comment:
Why do you pass typed array where the string is expected? I guess you can just call `.toString()` on it yourself before passing to `btoa`.